drivers/adc/adc_mcux_lpadc.c

```diff
@@ -423,6 +423,10 @@ static int mcux_lpadc_init(const struct device *dev)
 	RESET_PeripheralReset(kADC0_RST_SHIFT_RSTn);
 	CLOCK_AttachClk(kFRO_DIV4_to_ADC_CLK);
 	CLOCK_SetClkDiv(kCLOCK_DivAdcClk, 1);
+
+#elif defined(CONFIG_SOC_LPC55S36)
```

Comment:
we can do this in the soc.c instead to avoid adding to the technical debt related to the clocking in this file

Reply:
Lets do a separate PR to fix this completely... we will still need to have a change here to keep the #if/#else to work correctly for the 55s36 if we did the clock stuff in the soc.c